sonar-duplications/src/main/java/org/sonar/duplications/block/ByteArray.java

```diff
@@ -88,7 +89,15 @@ public ByteArray(int[] intArray) {
     int size = (bytes.length / 4) + (bytes.length % 4 == 0 ? 0 : 1);
     ByteBuffer bb = ByteBuffer.allocate(size * 4);
     bb.put(bytes);
-    bb.rewind();
+    // see https://github.com/mongodb/mongo-java-driver/commit/21c91bd364d38489e0bbe2e390efdb3746ee3fff
+    // The Java 9 ByteBuffer classes introduces overloaded methods with covariant return types for the following methods used by the driver:
+// Without casting, exceptions like this are thrown when executing on Java 8 and lower:
+
+   //  java.lang.NoSuchMethodError: java.nio.ByteBuffer.limit(I)Ljava/nio/ByteBuffer
+
+    //This is because the generated byte code includes the static return type of the method, which is not found on Java 8 and lower because
+    //the overloaded methods with covariant return types don't exist.
+    ((Buffer) bb).rewind();
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this unnecessary cast to "Buffer". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1905)